src/core/net/srp_client.cpp

```diff
@@ -976,6 +976,11 @@ Error Client::AppendServiceInstructions(Service &aService, Message &aMessage, In
 
     removing = ((aService.GetState() == kToRemove) || (aService.GetState() == kRemoving));
 
+    if (mSingleServiceMode.IsEnabled())
+    {
+        mSingleServiceMode.SetService(aService);
+    }
+
```

Comment:
Quick note: Just moving this `if()` block above, since we may exit early from line `VeriyOrExit(!removing)` if removing a service.